### PR TITLE
Create optimize fragment

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

* Create fragment for optimize command.
* Arrange hierarchy in Delta Lake ALTER TABLE EXECUTE section to match hierarchy in Hive and Iceberg. 
* Update reference links.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/optimize.fragment [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Optimize
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/optimize.fragment:1]: lowercase

File: docs/src/main/sphinx/connector/optimize.fragment [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ALTER TABLE test_table EXECUTE optimize(file_size_threshold => '10MB')
+```
+
+You can use a `WHERE` clause with the columns used to partition the table
+to filter which partitions are optimized:
+
+```sql
+ALTER TABLE test_partitioned_table EXECUTE optimize
+WHERE partition_key = 1
+```
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/optimize.fragment:28]: add empty line

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/hive.md [hunk 3]
```diff
@@ -715,37 +715,7 @@ The following operations are not supported when `avro_schema_url` is set:
 
 #### ALTER TABLE EXECUTE
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.md:717]: We should add the same intro here as we have in iceberg ..

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 4]
```diff
@@ -557,33 +557,7 @@ The connector supports the following [](/sql/alter-table) statements.
 
 ##### ALTER TABLE EXECUTE
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:559]: We should add the same intro here as we have in iceberg ..

Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:559]: We removed it from Delta Lake because it didn't flow with the new ALTER TABLE heading.

**Comment by mosabua** [comment]
Let me know when this is updated to look again @Jessie212

**Review by mosabua — APPROVED** [review]
Looks good now. Thank you.

**Pull request merged**